oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/prefetch/CacheWarming.java

```diff
@@ -48,6 +52,9 @@ public void prefetch(@NotNull Iterable<String> paths,
                 ids.add(id);
             }
         }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Prefetch {} nodes", ids.size());
+        }
```

Comment:
Just out of curiosity, does calling `LOG.isDebugEnabled()` actually helps performance in this case? My understanding is that `LOG.debug()` will only do a check if debug is enabled and returns at once if so, so the cost of calling that method is only the cost of evaluating the arguments. In this case, `ids.size()` simply returns an int, so it's very cheap (probably it's even inlined by the JIT). Or is there any other reason for the if guard?

Reply:
> does calling LOG.isDebugEnabled() actually helps performance in this case?

Probably it doesn't help currently, except if ids.size() is slow.
I just wanted to avoid any kind of risks, including the risk that someone might later add more data, that ids is null, things like that.
It's more following the pattern that all LOG.debug are behind LOG.isDebugEnabled()